addons/l10n_in/models/product_template.py

```diff
@@ -20,3 +25,21 @@ def _check_hsn_code_validation(self):
             if check_hsn and len(record.l10n_in_hsn_code) < int(minimum_hsn_len):
                 error_message = _("As per your HSN/SAC code validation, minimum %s digits HSN/SAC code is required.", minimum_hsn_len)
                 raise ValidationError(error_message)
+    @api.model
+    def get_hsn_suggestions(self, value):
+        response_json = {}
+        all_url_and_params = [
+            ('https://services.gst.gov.in/commonservices/hsn/search/qsearch', {'inputText': value, 'selectedType': 'byCode', 'category': 'null'}),
+            ('https://services.gst.gov.in/commonservices/hsn/search/qsearch', {'inputText': value, 'selectedType': 'byDesc', 'category': 'P'}),
+            ('https://services.gst.gov.in/commonservices/hsn/search/qsearch', {'inputText': value, 'selectedType': 'byDesc', 'category': 'S'}),
+        ]
```

Comment:
I think it would be best to move to the top of the file or create a const file.
 It's easy for someone to modify it, and if the government changes the URL/param, it's easy to make modifications. 

@jva-odoo will take the final decision.

Reply:
@ppr-odoo, I have made the changes, can you please have a look?